internal/controller/apiextensions/composite/composition_ptf.go

```diff
@@ -724,10 +741,29 @@ func RunFunction(ctx context.Context, fnio *iov1alpha1.FunctionIO, fn *v1.Contai
 		return nil, errors.Wrap(err, errDialRunner)
 	}
 
+	k8schainOpts := k8schain.Options{
+		ServiceAccountName: fnio.ServiceAccount,
+		Namespace:          fnio.Namespace,
+	}
+	for _, ips := range fn.ImagePullSecrets {
+		k8schainOpts.ImagePullSecrets = append(k8schainOpts.ImagePullSecrets, ips.Name)
+	}
+	keychain, err := k8schain.NewInCluster(ctx, k8schainOpts)
+	if err != nil {
+		if errors.Is(err, rest.ErrNotInCluster) {
+			keychain = nil
+		} else {
+			return nil, err
+		}
+	}
```

Comment:
```suggestion
	// If we're not in a cluster keychain will be nil.
	if err != nil && !errors.Is(err, rest.ErrNotInCluster) {
		return nil, errors.Wrap(err, errNewKeychain)
	}
```

I'd prefer this simpler implementation. I believe `keychain` should always be `nil` if it returns an error.

https://github.com/google/go-containerregistry/blob/v0.14.0/pkg/authn/k8schain/k8schain.go#L61